test/util/src/kafka/kafka_client.rs

```diff
@@ -60,23 +59,58 @@ impl KafkaClient {
             .await?;
 
         if res.len() != 1 {
-            return Err(format!(
+            return Err(anyhow::anyhow!(
                 "error creating topic {}: \
              kafka topic creation returned {} results, but exactly one result was expected",
                 self.topic,
                 res.len()
+            ));
+        }
+
+        if let Err((_, e)) = res[0] {
+            return Err(anyhow::anyhow!(
+                "error creating topic {}: {}",
+                self.topic,
+                e
+            ));
+        }
+
+        Ok(())
+    }
+
+    pub async fn delete_topic(&self) -> Result<(), anyhow::Error> {
```

Comment:
So I know in the past theres been pain around topic deletion being kind of doubly async and that's caused pain -- to the point that I have intentionally not added this type of functionality to the billing demo and instead forced users to do this manually if needed. not sure what your intended usecase is

Reply:
I can't remember what my intended use case was! Likely it was something like: run test, clean up. I'm happy to remove this